builds/misc/templates/mqtt-perf-setup.yaml

```diff
@@ -0,0 +1,118 @@
+# Template to set up test host (pull and start 'mqttd' container)
+# as well as publisher node and subscriber node (mqtt-benchmark tool writen in go).
+
+jobs:
+- job: setup_host
+  pool:
+    name: $(pool.name)
+    demands: mqtt-perf-tests-pub
+  displayName: Setup Host
+  steps:
+    - checkout: none
+    # AMD64 setup
+    - task: SSH@0
+      displayName: Cleanup running containers AMD64
+      inputs:
+        sshEndpoint: iotedge-mqtt-perf-amd64-conn
+        runOptions: commands        
+        commands: |
+          docker kill $(docker ps -q) 2>/dev/null
+      # ignore cleanup errors if no containers running
+      continueOnError: true 
+      condition: eq(variables['arch'], 'amd64')
```

Comment:
it is frustrating, but it looks like `sshEndpoint` (above) cannot be parametrized, b/c of that I had to do duplicate and fork the yaml for amd/arm and use `condition` instead.

Reply:
Check with @damonbarry to explore another approach, he implemented VM Proxy test in E2E pipeline, which agent also need to communicate with other machine.

Reply:
Is it possible to extract these steps as template and reuse it?

Reply:
Actually, you made me research more on it, and I found out that using "compile-time" variable `${{ variabes.<name> }}` solves the problem. So I got rid of the duplicate steps with conditions.

Reply:
I still see duplicate steps with conditions in this PR, are you going to remove duplicates in another PR?

Reply:
As discussed offline, I'll parametrize the templates and try to get rid of the conditional logic in Ssh task in a separate PR